app/models/miq_server/worker_management/monitor.rb

```diff
@@ -22,24 +22,19 @@ def monitor_workers
 
     MiqWorker.status_update_all
 
-    processed_worker_ids = []
-    processed_worker_ids += check_not_responding
-    processed_worker_ids += check_pending_stop
-    processed_worker_ids += clean_worker_records
+    check_not_responding
+    check_pending_stop
+    clean_worker_records
 
     # Monitor all remaining current worker records
-    processed_worker_ids += miq_workers.where(:status => MiqWorker::STATUSES_CURRENT_OR_STARTING).each do |worker|
-      # Check their queue messages for timeout
-      worker.validate_active_messages
+    miq_workers.where(:status => MiqWorker::STATUSES_CURRENT_OR_STARTING).each do |worker|
```

Comment:
@carbonin FYI, this commit: https://github.com/ManageIQ/manageiq/commit/e6cbfa8b4625147741919c723eacac540f056099 fixed the "can't modify frozen Hash" error.  If I change the above line to remove the `.where`, I can recreate that error for a simulated "exceeding memory worker" because it attempts to update a deleted worker from the cached association.  The `.where` make it run the query each time through the method.

Reply:
Oh, interesting. We may still want to make that explicit at some point though ... I wonder if we should do a `select` here and a `miq_workers.reload` at the start of the method? I guess we're really splitting hairs at that point though ...

Reply:
I think it's ok here as I'd expect a query like this to not be cached, so, there's no need to reload anything.  Relying on an association previously made it very unclear whether we had any expectations on caching.  I believe your change is much more explicit.  Maybe it's just me though?

Reply:
My concern is more clearly expressed in https://github.com/ManageIQ/manageiq/pull/19638/files#r357318923

I'm worried that someone removing the `where` for whatever reason wouldn't realize that we were relying on this line reloading the association for reasons other than this particular query.